core/src/main/java/org/apache/iceberg/SnapshotManager.java

```diff
@@ -159,6 +167,7 @@ public Snapshot apply() {
           return base.snapshot(targetSnapshotId);
         } else {
           // validate(TableMetadata) is called in apply(TableMetadata) after this apply refreshes the table state
+          requiresSnapshotCreation = true;
```

Comment:
I think this should be set when the operation is configured, not when it is applied.

Reply:
Can we reliably set requiresSnapshotCreation when operation is being configured, e.g a fast-forward can be converted to a cherry-pick only during apply which can result in snapshot creation, no?

Reply:
Answered here: https://github.com/apache/incubator-iceberg/pull/968#discussion_r415411038